src/ctelnet.cpp

```diff
@@ -929,12 +929,15 @@ QString cTelnet::decodeOption(const unsigned char ch) const
     }
 }
 
-std::pair<QString, int> cTelnet::getConnectionInfo() const
+std::tuple<QString, int, bool> cTelnet::getConnectionInfo() const
 {
+    // intentionally simplify connection state to a boolean
+    const bool connected = socket.state() == QAbstractSocket::ConnectedState;
```

Comment:
Can I make the observation that there may be a problem with the `getconnectionInfo()` API as you are proposing it after this PR: we currently do not (well, never *have*) monitored the connection process.

This means that this API does not handle a third (hopefully transitory) state of *connecting* - if a script/package makes use of this function and detects that the profile is *disconnected* it might immediate initiate a reconnection action - but Mudlet might actually be in the state of making the wanted connection - which means that calling `reconnect()` is **not** advisable then as it will call `(void) cTelnet::connectIf(...)` which will immediately abort an *in progress* connecting attempt!

Furthermore it might be that there is a problem (disconnected Ethernet cable for instance) that means it is not possible to connect - or perhaps a secure connection has been requested and the authentication has failed or even a name server failure. These are error condition which mean that it will be pointless trying to reconnect

Also we have not yet provided a timeout/retry mechanism that makes use of the `(int) Host::mTimeout` and `(int) Host::mRetries` members (which are present in the *oldest* commit in the Git History!).

All in all, simplifying this is not a good idea IMHO. It would be better to return the integer value of `(QAbstractSocket::SocketState) QAbstractSocket::state() const` and point out the values that may be seen:

0. Not connected
1. Performing a host name lookup (transitory, unless DNS fails)
2. Started establishing a connection (transitory, unless connection cannot be made - this is {or this and the prior one are} the stage(s) that should be monitored with a timer set from the `Host::mTimeout` value!)
3. Connected
4. Bound to an address and port (I don't think this will be seen with our usage of the socket, I think this is to do with Server rather than Client usage where the Server listens for connections)
5. For Qt internal use only (I don't think this will be seen with our usage of the socket)
6. About to close (data may still be waiting to be written) (transitory, on closing the connection)

Reply:
Most just want a simple toggle to know if they're connected or not, and they don't have an idea of how the connection process works either. It's not a good idea to pass all of this complexity onto the script author as you suggest - it won't be appreciated, it'll be frowned upon because it'll be too complicated.

If it actually causes problems in practice we can adjust it.

Reply:
> If it actually causes problems in practice we can adjust it.

Not without breaking ***backwards compatibility*** - as a boolean can only return `true`/`false` "connected" / "disconnected" - easier to just return an integer now and say `3`/`0` "connected"/"disconnected" and hint that other values might be seen during connection setup/teardown without documenting them.

Alternatively we could remap and return just two `int`eger values so that `0` = disconnected and `1` = connected and remap all the others to `0` as you have effectively done here - which would allow us to change that without changing the return value *type* if needed later.

Reply:
We can add another parameter for the exact connection state and go with a simple boolean for now, which is more Lua-like.

Reply:
So you think that `socket.state() == QAbstractSocket::ConnectedState` is the correct mapping for "connected/not connected" - sorry but I think that is going to bite us/connection package creators (perhaps if/when #3952 gets in) in the future - but I guess we can see how it works out...